src/app/grouping-key.service.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
```

Comment:
One thing we need to tackle soon is refactor all services that serve as a way to pass variable between components to use a better abstraction. 

We don't need to do this in this CL, we need to consider several alternatives first.
We can (i) use a single service to control simple passing variable logic like this, or (ii) implement a custom state management by ourselves, or (iii) use third party library like ngrx or ngxs. In react world, they are similar to redux.